snowflake_blueprints/upload_method_speed_tests.py

```diff
@@ -0,0 +1,369 @@
+from snowflake.connector.pandas_tools import pd_writer, write_pandas
+import pandas as pd
+from sqlalchemy import create_engine
+from snowflake.sqlalchemy import URL
+import argparse
+import shipyard_utils as shipyard
+import time
+import pyarrow as pa
+import pyarrow.parquet as pq
+import pyarrow.csv
+import dask.dataframe as dd
+
+import warnings
+warnings.simplefilter(action='ignore', category=UserWarning)
+warnings.filterwarnings(
+    action='ignore',
+    message='Dialect snowflake:snowflake will not make use of SQL compilation caching.*')
+
+DATA_SIZE = '50MB_WIDE'
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--username', dest='username', required=True)
+    parser.add_argument('--password', dest='password', required=True)
+    parser.add_argument('--account', dest='account', required=True)
+    parser.add_argument('--warehouse', dest='warehouse', required=False)
+    parser.add_argument('--database', dest='database', required=False)
+    parser.add_argument('--schema',
+                        dest='schema',
+                        default=None,
+                        required=False)
+    parser.add_argument('--source-file-name', dest='source_file_name',
+                        default='output.csv', required=True)
+    parser.add_argument('--source-folder-name',
+                        dest='source_folder_name', default='', required=False)
+    parser.add_argument(
+        '--insert-method',
+        dest='insert_method',
+        choices={
+            'fail',
+            'replace',
+            'append'},
+        default='append',
+        required=False)
+    args = parser.parse_args()
+
+    return args
+
+
+def start_timer(method_number):
+    print(f'Starting method {method_number}')
+    start = time.perf_counter()
+    print(start)
+    return start
+
+
+def end_timer(start, method_number):
+    finish = time.perf_counter()
+    print(finish)
+    print(
+        f'It took {finish - start} seconds to upload {DATA_SIZE} of data with method_{method_number}.')
+    return finish
+
+
+def create_table(source_full_path, table_name, db_connection):
+    try:
+        chunksize = 10000
+        for index, chunk in enumerate(
+                pd.read_csv(source_full_path, chunksize=chunksize)):
+            chunk.head(0).to_sql(
+                table_name,
+                con=db_connection,
+                if_exists="fail",
+                index=False)
+            # prevents a loop that was necessary for loading only a small chunk
+            # of the data in.
+            break
+    except BaseException as e:
+        print(e)
+
+
+def method_1(source_full_path,
```

Comment:
These names are bad - they don't mean anything to the user running the code, and require working knowledge of the code to know what's happening, i.e. looking at the documentation below to see. 
I highly suggest renaming to `method_put_copy_into_csv`, `method_put_copy_into_parquet`, etc.

Reply:
Wasn't anticipating anyone running this code. Just leaving it for posterity so I know _why_ the current updated method was chosen, when it's not the most logical on paper.